docs/Homebrew-linuxbrew-core-Maintainer-Guide.md

```diff
@@ -0,0 +1,269 @@
+conflict you have to replace the contents of `if OS.mac?` part up
+until `else` with the contents of the bottom part of the conflict
+("homebrew/master"). You also have to check if there are any obvious
+modifications that have to be made to the `else` part of the code that
+deals with non-Mac-related code.
+
+
+#### Finishing the merge
+
+Once all the conflicts have been resolved, a text editor will open
+with pre-populated commit message title and body:
+
+```text
+Merge branch homebrew/master into linuxbrew/master
+
+# Conflicts:
+#       Formula/git-lfs.rb
+#       Formula/gnutls.rb
+#       Formula/godep.rb
+```
+
+Leave the title of the message unchanged and uncomment all the
+conflicting files. Your final commit message should be:
+
+```text
+Merge branch homebrew/master into linuxbrew/master
+
+Conflicts:
+        Formula/git-lfs.rb
+        Formula/gnutls.rb
+        Formula/godep.rb
+```
+
+#### Submitting a PR
+
+The `merge-homebrew` command will create a pull-request for you, using `hub`.
+
+Once the PR successfully passes the tests and/or is approved by other
+Linuxbrew developers, you can finalize the merge with:
+
+```bash
+brew pull --clean <PR-NUMBER>
+git push origin master
+```
+
+The merge is now complete. Don't forget to update your GitHub fork by
+running `git push your-fork master`
+
+## Building bottles for updated formulae
+
+After merging changes, we must rebuild bottles for all the PRs that
+had conflicts.
+
+To do this, tap `Linuxbrew/homebrew-developer` and run the following
+command where the merge commit is `HEAD`:
+
+```sh
+for formula in brew find-formulae-to-bottle; do
+  brew build-bottle-pr --remote=<your-remote> $formula
+done
+```
+
+The `find-formulae-to-bottle` command outputs a list of formulae
+parsed from the merge commit body. It also performs some checks
+against the formulae:
+
+And it skips formulae if any of the following are true:
+- it doesn't need a bottle
+- it already has a bottle
+- the formula depends on macOS to build
+- the formula's tap is Homebrew/homebrew-core (the upstream macOS repo)
+- there is already an open PR for the formula's bottle
+- the current branch is not master
+
+By default, running this script won't output the errors. To see the
+errors, run `brew find-formulae-to-bottle --verbose` separate to the
+`for` loop above.
+
+The `build-bottle-pr` script creates a branch, adds `# Build a bottle
+for Linuxbrew` to the top of the formulae, pushes the branch to GitHub
+at the specified remote (default: `origin`), and raises a PR using `hub
+pull-request`.
+
+## Common build failures and how to handle them
+
+### Bottling errors
+
+## Handling `brew bump-formula-pr` PRs
+
+### Formulae that exist in Homebrew/homebrew-core
+
+The `brew bump-formula-pr` command will raise PRs against the Linux
+formula repo for upstream Mac formulae when running on Linux. This
+isn't how version bumps are done for _most_ formulae. Redirect users
+to raise their PRs in Homebrew/homebrew-core:
```

Comment:
Could link to https://github.com/Homebrew/brew/issues/6341 here to nudge people to help?